### PR TITLE
Add a static library for cuvs

### PR DESCRIPTION
**Pull request by benfred**
For the cuml integration, we need to be able to statically link to cuvs in order build the python wheels. This change adds a static target that lets us do that

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -465,11 +520,17 @@ if(NOT BUILD_CPU_ONLY)
     PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
     PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX> cuvs-cagra-search
   )
+
+  target_link_libraries(
+    cuvs_static
+    PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
+    PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
```
Comment (achirkin) [inline cpp/CMakeLists.txt:527]: Shouldn't this target also link `cuvs-cagra-search`?

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -430,12 +430,67 @@ add_library(
...
+)
+
+target_include_directories(
+  cuvs_static
+  PUBLIC "$<BUILD_INTERFACE:${DLPACK_INCLUDE_DIR}>"
+  INTERFACE "$<INSTALL_INTERFACE:include>"
+)
+
+# ensure CUDA symbols aren't relocated to the middle of the debug build binaries
+target_link_options(cuvs_static PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld")
```
Comment (achirkin) [inline cpp/CMakeLists.txt:488]: If you get a linker error here due to `cuvs-cagra-search` component, consider replacing this same as for the shared lib target:
```suggestion
target_link_options(cuvs_static PRIVATE $<HOST_LINK:${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld>)
```

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (benfred) [reply cpp/CMakeLists.txt:527]: yeah - I agree this should link cuvs-cagra-search (and tried in the previous commit) .   The error I was hitting was : 

```
  CMake Error in CMakeLists.txt:
    export called with target "cuvs_static" which requires target
    "cuvs-cagra-search" that is not in any export set.
```

https://github.com/rapidsai/cuvs/actions/runs/11155746120/job/31007185860

Will try your suggestion - thanks for this!

Comment (benfred) [reply cpp/CMakeLists.txt]: works great - thanks! applied in latest commit

Comment (achirkin) [reply cpp/CMakeLists.txt]: Do you think we still need the `cuvs-cagra-search` in the install export set? Theoretically, we shouldn't be exposing that as a library and nobody outside cuVS should need to link against it.

**Review by cjnolet — APPROVED** [review]

**Pull request merged**